contents/blog/clickhouse-vs-elasticsearch.md

```diff
@@ -0,0 +1,236 @@
+
+So why even compare them? Amid their differences, Elasticsearch and ClickHouse draw an interesting comparison because of their vastly different architecture, optimized for each of their respective goals. Comparing them is a good meditation on how various physical and virtual layouts can dramatically improve efficiency toward a specific efficiency goal. 
+
+## Background
+
+Sometimes, the relationship between an open-source tool and its lead developer is complicated; with ClickHouse, the relationship is straightforward and fundamentally no different from PostHog’s relationship with its open-source product. Elastic, meanwhile, has a complex history with open-source. 
+
+### What is Elasticsearch?
+
+Elasticsearch was originally released in 2010 under an open-source license. The premise behind Elasticsearch was that Apache Lucene, an open-source product designed to efficiently search JSON documents, needed better infrastructure for scaling. Apache Lucene made it easy to organize and search a series of JSON documents – such as human profiles; Elasticsearch made it easy to distribute those human profiles, which might be in the billions, across multiple locations, indexed both physically and virtually.
+
+Elasticsearch is considered a NoSQL database because it uses Apache Lucene – and by extension, JSON documents – as a primary store of data. Specifically, it is a Document-Store NoSQL database with a focus on searching and retrieving data. It is never used as the primary store of data. Sometimes Elasticsearch data stores are redundantly available in a more traditional database like PostgreSQL as Elasticsearch is only leveraged to improve search results. 
+
+In 2021, Elasticsearch abandoned its traditional Apache Open Source license in favor of a new license known as an Elastic license. It was a controversial move motivated by Elastic’s irritation with Amazon profiting off of Elasticsearch by operating a managed service without ever contributing to the codebase. Amazon forked the last version of open-source Elasticsearch into a new open-source project, OpenSearch. Similar to Elastic (and ClickHouse Inc.), Amazon launched a managed version of OpenSearch, titled OpenSearch Service.
+
+Elasticsearch’s new license allows developers to implement Elasticsearch themselves, but forbids cloud distributors from running a for-profit, managed Elasticsearch service. Most open-source advocates consider Elastic’s Elastic License not open-source; however, it would be unfair to Elastic to equate their solution’s transparency with a purely closed-source solution like Snowflake.
```

Comment:
```suggestion
Elasticsearch’s new license allows developers to implement Elasticsearch themselves, but forbids cloud distributors from running a for-profit, managed Elasticsearch service. Most open source advocates consider Elastic’s Elastic License not open source; however, it would be unfair to Elastic to equate their solution’s transparency with a purely closed-source solution like Snowflake.
```